src/winsign/sign.py

```diff
@@ -136,7 +138,10 @@ async def sign_file(
         if is_msix:
             winsign.makemsix.attach_signature(outfile, outfile, newsig)
         else:
-            write_signature(infile, outfile, newsig, certs)
+            if cafile is None or Path(cafile).is_file():
```

Comment:
I'm pretty sure we want `not Path...` here. Also, it might be better to do this check right near the top, before any work is done, so we can fail early.